crates/astria-sequencer/src/service/mempool.rs

```diff
@@ -170,5 +190,26 @@ async fn handle_check_tx<S: StateReadExt + 'static>(
         };
     };
 
+    // tx is valid, push to mempool
+    let priority = crate::mempool::TransactionPriority::new(
+        signed_tx.nonce(),
+        state
+            .get_account_nonce(Address::from_verification_key(signed_tx.verification_key()))
+            .await
+            .expect("can fetch account nonce"),
+    )
+    .expect(
+        "tx nonce is greater or equal to current account nonce; this was checked in \
+         check_nonce_mempool",
+    );
+    let mut mempool = mempool.lock().await;
+    mempool
+        .insert(signed_tx, priority)
+        .expect("priority transaction nonce and transaction nonce match, as we set them above");
     response::CheckTx::default()
 }
+
+async fn remove_from_mempool(tx_hash: &[u8; 32], mempool: Arc<Mutex<BasicMempool>>) {
```

Comment:
Very small nit: You could do the below because the hash is copy:
```suggestion
async fn remove_from_mempool(tx_hash: [u8; 32], mempool: Arc<Mutex<BasicMempool>>) {
```

Reply:
removed this method